starter_code/src/movies.js

```diff
@@ -4,9 +4,40 @@
 
 // Get the average of all rates with 2 decimals 
 
+function ratesAverage (movies) {
+    var myAverage = 
+        movies.reduce(function (acc, oneRate) {
+        return acc + parseFloat(oneRate.rate);
+}, 0);
+return(myAverage / movies.length);
+}
+
+ratesAverage (movies);
+
 
 // Get the average of Drama Movies
 
+// new array with only dramas
+function  dramaMoviesThing (myArray) {
+    var dramaMovies = 
+      myArray.filter(function (oneMovie) {
+      return oneMovie.genre.includes('Drama');
+    });
+    return dramaMovies;
+  }
+
+  // get the average
+  function dramaMoviesRate (myArray) {
+  var myDramaArray =  dramaMoviesThing(myArray);
+  var myDramaAverage = 
+    myDramaArray.reduce(function(acc, oneRate){
+      return acc + parseFloat(oneRate.rate);
+    }, 0);
+    return (myDramaAverage / myDramaArray.length);
+  }
```

Comment:
Yes, good job !
since the second part of that function is exactly the same as the ratesAverage function, you chan just call it on your drama array : 
```
 function dramaMoviesRate (myArray) {
  var myDramaArray =  dramaMoviesThing(myArray);
    return ratesAverage(myDramaArray);
  }
```
To get rid of the error you still have on Jasmine, you can add this condition : 
```
 function dramaMoviesRate (myArray) {
  var myDramaArray =  dramaMoviesThing(myArray);
  if (myDramaArr.length > 0) {
    return ratesAverage(myDramaArray);
  }
  }
```
(The very last error on Jasmin doesn't make sense to me, but you can add this condition in the ratesAverage function to get rid of it : 
    if (curr.rate === "") {
      curr.rate = 0;
    }
)